source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -184,8 +231,8 @@ void TextureBaker::optimizeBakedTextures(NodePtr shader)
             }
         }
 
-        // Check for uniform outputs.
-        if (outputIsUniform)
+        // Add a constant for each uniform if optimizing out uniforms
```

Comment:
@jstone-lucasfilm , this is the refined logic to handle when optimization is disabled.
Here the restricted logic is to only create constants if the image is a uniform and optimization to constants
is enabled. This also fixes cases where neigther a constant value, nor an image was being created if
optimization was disabled.